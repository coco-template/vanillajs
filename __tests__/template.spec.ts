/* eslint-disable import/no-extraneous-dependencies, no-console */

// native
const fs = require('fs');
const path = require('path');
// external
const YAML = require('yamljs');
const handlebars = require('handlebars');
// scope
const coco = path.resolve(process.cwd(), 'coco.yml');
const { templates } = YAML.load(coco);
const context = {
  scope: 'coco-template',
  package: 'web-package',
  author: 'huang.jian',
  email: 'hjj491229492@hotmail.com',
  provider: 'github.com',
};

describe('template', () => {
  it('should render standard workflow', () => {
    const reflection = templates.map((template: string) => {
      const hbs = path.resolve(process.cwd(), `${template}.hbs`);
      const options = { encoding: 'utf-8' };

      return fs.readFileSync(hbs, options);
    });

    reflection.forEach((hbs: string) => {
      expect(handlebars.compile(hbs)(context)).toMatchSnapshot();
    });
  });
});
