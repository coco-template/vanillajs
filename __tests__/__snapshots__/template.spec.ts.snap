// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template should render standard workflow 1`] = `
"{
  \\"name\\": \\"web-package\\",
  \\"version\\": \\"0.2.0\\",
  \\"description\\": \\"A lite seed to develop with typescript.\\",
  \\"module\\": \\"es/index.js\\",
  \\"main\\": \\"bundle/index.common.js\\",
  \\"typings\\": \\"native/index.d.ts\\",
  \\"files\\": [\\"es\\", \\"lib\\", \\"native\\"],
  \\"scripts\\": {
    \\"prepublish\\": \\"npm-run-all -s clean compile\\",
    \\"prepush\\": \\"npm test\\",
    \\"precommit\\": \\"pretty-quick --staged\\",
    \\"dev\\": \\"webpack-serve\\",
    \\"clean\\": \\"rm -rf native es lib\\",
    \\"compile\\": \\"npm-run-all -s compile:*\\",
    \\"compile:native\\": \\"tsc --declaration --outDir native\\",
    \\"compile:esm\\": \\"babel native --source-maps --out-dir es\\",
    \\"compile:es5\\":
      \\"babel native --source-maps --out-dir lib --plugins=transform-es2015-modules-commonjs\\",
    \\"test\\": \\"jest -c jest.config.js --coverage\\",
    \\"test:watch\\": \\"jest -c jest.config.js --watch\\"
  },
  \\"repository\\": {
    \\"type\\": \\"git\\",
    \\"url\\": \\"git+https://github.com/coco-template/web-package.git\\"
  },
  \\"keywords\\": [\\"typescript\\", \\"package\\"],
  \\"author\\": \\"huang.jian <hjj491229492@hotmail.com>\\",
  \\"contributors\\": [
    {
      \\"name\\": \\"huang.jian\\",
      \\"email\\": \\"hjj491229492@hotmail.com\\"
    },
    {
      \\"name\\": \\"yang.yuncai\\",
      \\"email\\": \\"383235388@qq.com\\"
    }
  ],
  \\"license\\": \\"MIT\\",
  \\"bugs\\": {
    \\"url\\": \\"https://github.com/coco-template/web-package/issues\\"
  },
  \\"homepage\\": \\"https://github.com/coco-template/web-package#readme\\",
  \\"dependencies\\": {},
  \\"devDependencies\\": {
    \\"@coco-platform/webpack-preset\\": \\"^0.2.1\\",
    \\"@types/jest\\": \\"^22.2.3\\",
    \\"babel-cli\\": \\"^6.26.0\\",
    \\"babel-plugin-transform-class-properties\\": \\"^6.24.1\\",
    \\"babel-plugin-transform-es2015-modules-commonjs\\": \\"^6.26.2\\",
    \\"babel-plugin-transform-runtime\\": \\"^6.23.0\\",
    \\"babel-preset-env\\": \\"^1.7.0\\",
    \\"babel-preset-stage-3\\": \\"^6.24.1\\",
    \\"eslint\\": \\"^4.2.0\\",
    \\"eslint-config-airbnb-base\\": \\"^12.1.0\\",
    \\"eslint-plugin-import\\": \\"^2.12.0\\",
    \\"husky\\": \\"^0.14.3\\",
    \\"jest\\": \\"^22.4.4\\",
    \\"npm-run-all\\": \\"^4.0.2\\",
    \\"prettier\\": \\"^1.12.1\\",
    \\"pretty-quick\\": \\"^1.5.0\\",
    \\"ts-jest\\": \\"^22.4.6\\",
    \\"tslint\\": \\"^5.10.0\\",
    \\"typescript\\": \\"^2.8.3\\",
    \\"webpack-serve\\": \\"^1.0.2\\"
  }
}
"
`;

exports[`template should render standard workflow 2`] = `
"# web-package

![Build Status](https://img.shields.io/travis/coco-template/web-package/master.svg?style=flat)
[![Coverage Status](https://coveralls.io/repos/github/coco-template/web-package/badge.svg?branch=master)](https://coveralls.io/github/coco-template/web-package?branch=master)
![Package Dependency](https://david-dm.org/coco-template/web-package.svg?style=flat)
![Package DevDependency](https://david-dm.org/coco-template/web-package/dev-status.svg?style=flat)

A lite seed to develop package with typescript.

## Attention

* use typescript for test suits.
* avoid below code for \`babel-plugin-transform-runtime\` weired issue:

\`\`\`typescript
export * from './src/something';
\`\`\`

## Usage

\`\`\`shell
# start develop server
npm run dev;

# unit test
npm run test;

# production compile
npm run compile;
\`\`\`

## Contact

hjj491229492@hotmail.com

## License

MIT
"
`;
